truffle/src/main/java/org/jruby/truffle/nodes/core/RegexpNodes.java

```diff
@@ -54,17 +56,17 @@ public static Object matchCommon(RubyContext context, DynamicObject regexp, Dyna
         assert RubyGuards.isRubyRegexp(regexp);
         assert RubyGuards.isRubyString(source);
 
-        final ByteList sourceByteList = StringOperations.getByteListReadOnly(source);
+        final Rope sourceRope = StringOperations.rope(source);
 
-        final ByteList bl = Layouts.REGEXP.getSource(regexp);
-        final Encoding enc = checkEncoding(regexp, StringOperations.getCodeRangeableReadOnly(source), true);
-        final ByteList preprocessed = RegexpSupport.preprocess(context.getRuntime(), bl, enc, new Encoding[] { null }, RegexpSupport.ErrorMode.RAISE);
+        final Rope regexpSourceRope = Layouts.REGEXP.getSource(regexp);
+        final Encoding enc = checkEncoding(regexp, sourceRope, true);
+        final ByteList preprocessed = RegexpSupport.preprocess(context.getRuntime(), regexpSourceRope.getUnsafeByteList(), enc, new Encoding[] { null }, RegexpSupport.ErrorMode.RAISE);
```

Comment:
Ideally I want to pull this out of `Rope` entirely, I think. But I've made a note :-)